src/service/history-impl.js

```diff
@@ -463,9 +527,34 @@ export class HistoryBindingNatural_ {
     stackIndex = Math.max(stackIndex, this.startIndex_);
     return this.whenReady_(() => {
       return this.back_(this.stackIndex_ - stackIndex + 1);
+    }).then(newStackIndex => {
+      return this.mergeStateUpdate_(this.getState_(), {
+        stackIndex: newStackIndex,
+      });
     });
   }
 
+  /** @override */
+  replace(opt_stateUpdate) {
+    return this.whenReady_(() => {
+      const updatedState = this.mergeStateUpdate_(
+          this.getState_(),
+          opt_stateUpdate || {});
+      this.historyReplaceState_(updatedState, '',
+          '#' + (updatedState.fragment || ''));
```

Comment:
Minor: Avoid updating the URL with an empty fragment? Also, let's pass `undefined` to the useless `title` param.

```js
const {fragment} = updatedState;
this.historyReplaceState_(updatedState, /* title */ undefined, 
    fragment ? `#${fragment}` || undefined);
```

Reply:
Done.